src/client/array/dc_array.c

```diff
@@ -2129,10 +2130,9 @@ punch_extent(daos_handle_t oh, daos_handle_t th, daos_size_t dkey_val, daos_off_
 	io_arg->iods	= iod;
 	io_arg->sgls	= sgl;
 
-	rc = tse_task_register_comp_cb(io_task, free_io_params_cb, &params,
-				       sizeof(params));
+	rc = tse_task_register_comp_cb(io_task, free_io_params_cb, &params, sizeof(params));
 	if (rc)
-		D_GOTO(err, rc);
+		D_GOTO(free_reqs, rc);
```

Comment:
It appears `D_GOTO(err, rc);`  below should also be `D_GOTO(free_reqs, rc);`

Reply:
No, because there the free callback has been registered so that would result in a double-free.

Reply:
> the free callback has been registered

Where?

Reply:
Do you mean on L2139?  This line we're commenting on is registering the free callback to free params so they'll be freed as part of tse_task_complete.

As the code stands in master I was seeing a double-free although this PR does not exercise this code, as part of this work I added https://daosio.atlassian.net/browse/DAOS-13397 to create a new test for copying to an existing target which will extend coverage to this function.